doc/configuration/Configuration.md

```diff
@@ -0,0 +1,319 @@
+
+```"FileFormat": ""```
+*Format of log message output. If blank, FileFormat = "[%D %T] [%L] (%S) %M", where: 
+  
+    %T		Time (15:04:05 MST) 
+    %t		Time (15:04) 
+    %D		Date (2006/01/02) 
+    %d		Date (01/02/06) 
+    %L		Level (FNST, FINE, DEBG, TRAC, WARN, EROR, CRIT) 
+    %S		Source 
+    %M		Message*
+
+```"FileLocation": ""```
+*Directory to which log files are written. If blank, log files write to ./logs/mattermost, which writes logs to mattermost.log. Log rotation is enabled and every 10,000 lines of log information is written to new files stored in the same directory, for example mattermost.2015-09-23.001, mattermost.2015-09-23.002, and so forth.*
+
+
+### Rate Limit Settings
+___
+Settings to enable API rate limiting and configure requests per second, user sessions and variables for rate limiting. Changing properties in this section will require a server restart before taking effect.
+
+```"EnableRateLimiter": true```
+*”true”: APIs are throttled at rate specified by PerSec.*
+
+```"PerSec": 10```
+*Throttle API at this number of requests per second if EnableRateLimiter=true.*
+
+```"MemoryStoreSize": 10000```
+*Maximum number of user sessions connected to the system as determined by VaryByRemoteAddr and VaryByHeader variables.*
+
+```"VaryByRemoteAddr": true```
+*"true": Rate limit API access by IP address.*
+
+```"VaryByHeader": ""```
+*Vary rate limiting by HTTP header field specified (e.g. when configuring Ngnix set to "X-Real-IP", when configuring AmazonELB set to "X-Forwarded-For").*
+
+
+### Privacy Settings
+___
+Settings to configure the name and email privacy of users on your system.  
+
+```"ShowEmailAddress": true```
+*“true”: Show email address of all users; "false": Hide email address of users from other users in the user interface, including team owners and team administrators. This can be used when system is set up for managing teams where some users choose to keep their contact information private.*
+
+```"ShowFullName": true```
+*”true”: Show full name of all users; “false”: hide full name of users from other users including team owner and team administrators.*
+
+```"EnableDiagnostic": false```
+*”true”: Server will periodically send error and diagnostic information to Mattermost.* 
+
+### GitLab Settings
+___
+Settings to configure account and team creation using GitLab OAuth.
+
+```"Enable": false```
+*true: Allow team creation and account signup using GitLab OAuth. To configure, input the Secret and Id variables.* 
+
+```"Secret": ""```
+*Obtain this value by logging into your GitLab account. Go to Applications -> Profile Settings. Enter Redirect URLs "<your-mattermost-url>/login/gitlab/complete" (example: http://localhost:8065/login/gitlab/complete) and "<your-mattermost-url>/signup/gitlab/complete".*
```

Comment:
You also need to enter an app name (e.g. Mattermost) on the GitLab page, not sure if it's worth mentioning here